main.go

```diff
@@ -234,15 +242,20 @@ func serveHTTP(rw http.ResponseWriter, r *http.Request) {
 		promHandler.ServeHTTP(rw, r)
 	case "/", "/query":
 		var err error
+
+		proxySettings := proxySettings.Load().(*config.Proxy)
+		proxyHandler := NewProxyHandler(proxySettings)
+		remoteAddr := proxyHandler.GetRemoteAddr(r)
```

Comment:
you extract the remoteAddr but I don't see where you do the logic r.RemoteAddr = remoteAddr
I'm not an expert but I guess this logic is needed given the pb, cf a lib doing the same as what we do https://github.com/gorilla/handlers/blob/v1.5.1/proxy_headers.go#L43
I might have loose it because I'm sure this logic was in your code 7 days ago